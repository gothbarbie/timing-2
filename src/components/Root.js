import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'

import defaultStore from '../store/configureStore'

const Root = ({ store, children }) => (
  <Provider store={store ? store : defaultStore}>{children}</Provider>
)

Root.propTypes = {
  store: PropTypes.shape({}),
  children: PropTypes.node,
}

export default Root
