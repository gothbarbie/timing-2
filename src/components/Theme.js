import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
