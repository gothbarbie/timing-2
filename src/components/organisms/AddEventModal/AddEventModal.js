import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { connect } from 'react-redux'

import { openModal } from '../../../actions/modalActions'

import Button from '../../atoms/Button'

import LabeledInput from '../../molecules/LabeledInput'
import LabeledSelect from '../../molecules/LabeledSelect'
import LabeledTags from '../../molecules/LabeledTags'

const TwoColumns = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 3rem;
  margin: 3rem 0;
`

const CategorySection = styled.div`
  display: grid;
  grid-template-columns: 6fr 1fr;
  grid-gap: 1rem;
  margin: 3rem 0;
  align-items: flex-end;
`

const AlignRight = styled.div`
  margin-top: 3rem;
  display: flex;
  justify-content: flex-end;
  width: 100%;
`

const AddEventModal = ({ openModal }) => (
  <section>
    <LabeledInput label="Title" name="title" />
    <TwoColumns>
      <LabeledInput
        label="Start"
        name="start"
        type="date"
        onChange={() => console.log('changed')}
      />
      <LabeledInput
        label="End"
        name="end"
        type="date"
        onChange={() => console.log('changed')}
      />
    </TwoColumns>
    <CategorySection>
      <LabeledSelect
        label="Category"
        name="category"
        onChange={() => console.log('changed')}
        options={[{ key: 'Test', name: 'Test Name', value: 'TestValue' }]}
      />
      <Button
        fullWidth
        onClick={() => openModal({ content: 'editCategories' })}
        primary
      >
        Edit
      </Button>
    </CategorySection>
    <TwoColumns>
      <LabeledInput
        label="Add tags"
        name="add-tags"
        onChange={() => console.log('changed')}
      />
      <LabeledTags
        label="Used recently"
        onClick={id => console.log('added', id)}
        tags={[{ blue: true, name: 'unit test', id: '1' }]}
      />
    </TwoColumns>
    <LabeledTags
      label="Selected tags"
      onClick={id => console.log('deleted', id)}
      tags={[{ blue: false, name: 'story-125', id: '3' }]}
    />
    <AlignRight>
      <Button primary>Create</Button>
    </AlignRight>
  </section>
)

AddEventModal.propTypes = {
  openModal: PropTypes.func.isRequired,
  tags: PropTypes.arrayOf(
    PropTypes.shape({
      blue: PropTypes.bool.isRequired,
      name: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,
}

const mapStateToProps = ({ tags }) => ({
  tags,
})

const mapDispatchToProps = {
  openModal,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddEventModal)
