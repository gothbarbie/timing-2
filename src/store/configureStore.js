import { combineReducers, createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension'

import categoryReducer from '../reducers/categoryReducer'
import categoriesFilterReducer from '../reducers/categoriesFilterReducer'
import modalReducer from '../reducers/modalReducer'
import weekReducer from '../reducers/weekReducer'
import tagsReducer from '../reducers/tagsReducer'
import timeSlotsReducer from '../reducers/timeSlotsReducer'
import databaseErrorReducer from '../reducers/databaseErrorReducer'

const rootReducer = combineReducers({
  categories: categoryReducer,
  categoriesFilter: categoriesFilterReducer,
  modal: modalReducer,
  tags: tagsReducer,
  week: weekReducer,
  timeSlots: timeSlotsReducer,
  error: databaseErrorReducer,
})

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
)

export default store
