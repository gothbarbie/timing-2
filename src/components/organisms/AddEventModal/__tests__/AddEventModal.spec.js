import React from 'react'
import { render } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faSearch,
  faPencilAlt,
  faTimes,
} from '@fortawesome/free-solid-svg-icons'

import { theme } from '../../../Theme/Theme'

import AddEventModal from '../AddEventModal'
import Root from 'Root'
import configureStore from 'store'

library.add(faSearch, faPencilAlt, faTimes)

const store = configureStore()

describe('generic/organisms/AddEventModal', () => {
  it('renders', () => {
    const component = render(
      <Root store={store}>
        <ThemeProvider theme={theme}>
          <AddEventModal />
        </ThemeProvider>
      </Root>
    )

    expect(component).toMatchSnapshot()
  })
})
