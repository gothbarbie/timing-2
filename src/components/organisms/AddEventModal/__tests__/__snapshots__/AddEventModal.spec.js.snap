// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/organisms/AddEventModal renders 1`] = `
<section>
  <div
    class="sc-EHOje fpqoIO"
  >
    <label
      class="sc-bZQynM snbku"
      for="title"
    >
      Title
    </label>
    <input
      class="sc-ifAKCX cYWJSo"
      name="title"
      type="text"
    />
  </div>
  <div
    class="sc-jTzLTM eaHDAf"
  >
    <div
      class="sc-EHOje fpqoIO"
    >
      <label
        class="sc-bZQynM snbku"
        for="start"
      >
        Start
      </label>
      <input
        class="sc-ifAKCX cYWJSo"
        name="start"
        type="date"
      />
    </div>
    <div
      class="sc-EHOje fpqoIO"
    >
      <label
        class="sc-bZQynM snbku"
        for="end"
      >
        End
      </label>
      <input
        class="sc-ifAKCX cYWJSo"
        name="end"
        type="date"
      />
    </div>
  </div>
  <div
    class="sc-fjdhpX kvktnV"
  >
    <div
      class="sc-dnqmqq iSdCun"
    >
      <label
        class="sc-iwsKbI ishnVy"
        for="category"
      >
        Category
      </label>
      <select
        class="sc-htoDjs fTgUHt"
        name="category"
      >
        <option
          class="sc-gzVnrw ibZhkv"
          value="TestValue"
        >
          TestValue
        </option>
      </select>
    </div>
    <button
      class="sc-htpNat cdaQKa"
    >
      Edit
    </button>
  </div>
  <div
    class="sc-jTzLTM eaHDAf"
  >
    <div
      class="sc-EHOje fpqoIO"
    >
      <label
        class="sc-bZQynM snbku"
        for="add-tags"
      >
        Add tags
      </label>
      <input
        class="sc-ifAKCX cYWJSo"
        name="add-tags"
        type="text"
      />
    </div>
    <div
      class="sc-gqjmRU foEULD"
    >
      <label
        class="sc-VigVT iIvxkR"
      >
        Used recently
      </label>
      <button
        class="sc-gZMcBi daxmKN"
      >
        unit test
      </button>
    </div>
  </div>
  <div
    class="sc-gqjmRU foEULD"
  >
    <label
      class="sc-VigVT iIvxkR"
    >
      Selected tags
    </label>
    <button
      class="sc-gZMcBi kcpxAK"
    >
      story-125
    </button>
  </div>
  <div
    class="sc-jzJRlG jumTEP"
  >
    <button
      class="sc-htpNat elmvaw"
    >
      Create
    </button>
  </div>
</section>
`;
